### DISCUSSION TITLE
Render count is high with useReducerAtom

### DISCUSSION THREAD

**Discussion by attiqeurrehman** [pmndrs/jotai#1524] [Q&A] [answered]
Hi,

I have been trying out the library recently and a lot of our code base is on reducers I want to see the viability of using `useReducerAtom` with since it's already supported by the library. 

I have made [changes ](https://codesandbox.io/s/affectionate-haibt-g65lce?file=/src/App.js:518-532)in the reducer example from the [react docs](https://beta.reactjs.org/learn/extracting-state-logic-into-a-reducer#step-3-use-the-reducer-from-your-component). But resultant render count is still higher. 

Maybe I am missing something here?

**Comment by dai-shi** [comment, marked as answer]
https://codesandbox.io/s/solitary-darkness-83w3dz
If you count commits (useEffect), it should be more expected result.
This is what we call extra-renders-without-commits behavior, and it's intentional.

(btw, i'm currently working on a new api, which changes this behavior. #1514 )

**Reply by attiqeurrehman** [reply]
Thank you so much for your quick reply and answer.
